decisions/wc_wonder_decisions.txt

```diff
@@ -0,0 +1,210 @@
+decisions = {
+	# old god retinue
+	recruit_old_god_guard = {
+		only_independent = yes
+		is_high_prio = yes
+		ai_check_interval = 12
+
+		potential = {
+			is_playable = yes
+			NOT = { has_earmarked_regiments = old_god_guard }
+			any_realm_wonder_upgrade = {
+				OR = {
+					has_wonder_upgrade = upgrade_old_god_cthun_1
+					has_wonder_upgrade = upgrade_old_god_cthun_2
+					has_wonder_upgrade = upgrade_old_god_nzoth_1
+					has_wonder_upgrade = upgrade_old_god_nzoth_2
+					has_wonder_upgrade = upgrade_old_god_yogg_saron_1
+					has_wonder_upgrade = upgrade_old_god_yogg_saron_2
+				}
+				is_active = yes
+			}
+		}
+		allow = {
+		}
+		effect = {
+			random_realm_wonder_upgrade = {
+				limit = {
+					OR = {
+						has_wonder_upgrade = upgrade_old_god_cthun_1
+						has_wonder_upgrade = upgrade_old_god_cthun_2
+						has_wonder_upgrade = upgrade_old_god_nzoth_1
+						has_wonder_upgrade = upgrade_old_god_nzoth_2
+						has_wonder_upgrade = upgrade_old_god_yogg_saron_1
+						has_wonder_upgrade = upgrade_old_god_yogg_saron_2
+					}
+					is_active = yes
+				}
+				wonder = { save_event_target_as = the_wonder }
+			}
+			if = {
+				limit = { event_target:the_wonder = { has_wonder_upgrade = upgrade_old_god_cthun_1 } }
+				event_target:the_wonder = {
+					show_scope_change = no
+					location = {
+						show_scope_change = no
+						PREVPREV = {
+							show_scope_change = no 
+							spawn_unit = {
+								owner = ROOT
+								province = PREV
+								home = PREV
+								earmark = old_god_guard
+								troops = {
+									pikemen = { 480 480 }
+									knights = { 240 240 }
+									swarm = { 3600 3600 }
+								}
+								reinforces = yes
+								attrition = 0.2
+								maintenance_multiplier = 0
+							}
```

Comment:
It can be replaced by a scripted effect to make it more compact and easy to review.

Reply:
It's needed. `upgrade_old_god_nzoth_2` spawns x2 troops of `upgrade_old_god_nzoth_1`

Reply:
yes but they are one unit not two

Reply:
> yes but they are one unit not two

The real question is why so? x2 lines without any reason.

Reply:
so that you cant split them into 2 units

Reply:
> so that you cant split them into 2 units

What for?

Reply:
cause its one unit

Reply:
So, there's no reason?

Reply:
its one unit and youre not supposed to split to siege two places at once or whatever

Reply:
> its one unit and youre not supposed to split to siege two places at once or whatever

I'm not supposed to check 100 lines that can be replaced by 8 lines too.

Reply:
Especially if I don't see any other reason except annoying the players.

Reply:
so scripted effects dont need to be checked?

Reply:
> so scripted effects dont need to be checked?

I didn't say it. I need to check less.

Reply:
so its not 8 lines

Reply:
the unit spawns as one because thats how it works in vanilla and its less confusing in general for anyone trying to edit the file

Reply:
More lines = less confusing. Got it.

Reply:
you dont need to find the file that has the effect or even google for what it means. so someone that doesnt know how it works will have easier time editing it

Reply:
> you dont need to find the file that has the effect or even google for what it means. so someone that doesnt know how it works will have easier time editing it

Except... we have all the unit spawns in `wc_unit_scripted_effects`.